src/Analysis/Ast/Impl/Analyzer/Symbols/SymbolCollector.cs

```diff
@@ -101,12 +101,15 @@ internal sealed class SymbolCollector : PythonWalker {
             var declaringType = fd.Parent != null && _typeMap.TryGetValue(fd.Parent, out var t) ? t : null;
             var existing = _eval.LookupNameInScopes(fd.Name, LookupOptions.Local);
 
-            switch (existing?.MemberType) {
-                case PythonMemberType.Method:
-                case PythonMemberType.Function:
-                case PythonMemberType.Property:
-                    ReportRedefinedFunction(fd, existing as PythonType);
-                    break;
+            // lambdas have the same function name, so don't count them as redefinitions
+            if (!string.IsNullOrEmpty(fd.Name) && !fd.IsLambda && !IsRedefinedByDecorator(fd)) {
```

Comment:
On a second reading, I think it'd be best to do this:

```
if (existing != null
    && !fd.IsLambda
    && !string.IsNullOrEmpty(fd.Name)
    && !IsRedefinedByDecorator(fd)) {
    // ...
}
```

As for the most part, there aren't going to be any redefinitions, and `existing` will be null. As written now with the switch case's `existing?` check, the null check happens last, even though it's the most likely.

Reply:
Good insight, I'll add it.